File: bulk-email-campaign-system/src/main/java/com/enterprise/emailcampaign/service/CampaignService.java
package com.enterprise.emailcampaign.service;

import com.enterprise.emailcampaign.model.dto.request.CampaignRequest;
import com.enterprise.emailcampaign.model.dto.response.CampaignResponse;
import com.enterprise.emailcampaign.model.dto.response.DeliveryLogResponse;

import java.util.List;

/**
 * Service contract for campaign lifecycle management.
 */
public interface CampaignService {

    CampaignResponse createCampaign(CampaignRequest request);

    CampaignResponse getCampaignById(Long id);

    List<CampaignResponse> getAllCampaigns();

    CampaignResponse updateCampaign(Long id, CampaignRequest request);

    void deleteCampaign(Long id);

    CampaignResponse scheduleCampaign(Long id);

    /**
     * Triggers immediate execution of a campaign regardless of scheduled time.
     */
    void executeCampaign(Long id);

    List<DeliveryLogResponse> getDeliveryLogs(Long campaignId);
}

File: email-campaign/src/services/campaignService.js
import api from '../api/api'

export const getAllCampaigns   = ()           => api.get('/campaigns')
export const getCampaignById   = (id)         => api.get(`/campaigns/${id}`)
export const createCampaign    = (data)       => api.post('/campaigns', data)
export const updateCampaign    = (id, data)   => api.put(`/campaigns/${id}`, data)
export const deleteCampaign    = (id)         => api.delete(`/campaigns/${id}`)
export const scheduleCampaign  = (id)         => api.post(`/campaigns/${id}/schedule`)
export const executeCampaign   = (id)         => api.post(`/campaigns/${id}/execute`)
export const getDeliveryLogs   = (id)         => api.get(`/campaigns/${id}/delivery-logs`)
